### PR TITLE
docs: adding code host connections on local dev env

### PR DESCRIPTION
**Pull request by limitedmage**
There are currently no docs on how to enable code host connections on local dev environment, so adding it here.

### CODE REVIEW

**Review by mrnugget — APPROVED** [review]
Thanks! cc @sourcegraph/core-application in case there's something missing here

**Comment by ryanslade** [comment]
Thanks for adding this 👍 

If a user has the `AllowUserExternalServicePrivate` tag they will also be able to add public code.

It can also be set at the site level via site config, see:
https://github.com/sourcegraph/sourcegraph/blob/e34980d87e000c0437313286580fa873b9b51150/schema/site.schema.json#L569-L574

**Comment by mrnugget** [comment]
> It can also be set at the site level via site config, see:

Does anything speak against adding this to https://github.com/sourcegraph/dev-private/blob/master/enterprise/dev/site-config.json and the

**Comment by ryanslade** [comment]
> > It can also be set at the site level via site config, see:
> 
> Does anything speak against adding this to https://github.com/sourcegraph/dev-private/blob/master/enterprise/dev/site-config.json and the

I think it's safe to make `all` the default now. @artemruts does that sound good?

**Comment by limitedmage** [comment]
In that case, if we enable it via site config for localhost, do we still want these docs added? I think they wouldn't be necessary at that point.

**Comment by mrnugget** [comment]
Created these two PRs:

https://github.com/sourcegraph/sourcegraph/pull/25219
https://github.com/sourcegraph/dev-private/pull/31

I think having the docs additionally doesn't hurt, especially since they explain the SQL way to do this. But with these two PRs and the site-config option, that should probably also be mentioned in the docs.

**Comment by limitedmage** [comment]
OK I've updated the docs, let me know if that looks OK to check in :)

**Comment by artemruts** [comment]
> > > It can also be set at the site level via site config, see:
> > 
> > 
> > Does anything speak against adding this to https://github.com/sourcegraph/dev-private/blob/master/enterprise/dev/site-config.json and the
> 
> I think it's safe to make `all` the default now. @artemruts does that sound good?

I think it's a good idea! At this point, everyone should have access and be able to test.

**Review by mrnugget — APPROVED** [review]
LGTM!

File: doc/dev/background-information/code-host.md [hunk 1]
```diff
@@ -5,11 +5,27 @@ Connecting to a code host is required to develop and test certain features, incl
...
+## Enabling for specific users
+
+To enable code host connections for a user, two flags need to be set:
 `AllowUserExternalServicePublic` and `AllowUserExternalServicePrivate`.
 These flags may not be set when testing locally. You can run the following
 SQL command to set the flags for all users in your local test instance:
 
 ```sql
-update users set tags = '{AllowUserExternalServicePublic,AllowUserExternalServicePrivate}';
+update users
```
Comment (ryanslade) [inline doc/dev/background-information/code-host.md:28]:
```suggestion
update users set tags = array_append(tags, 'AllowUserExternalServicePrivate') where not ('AllowUserExternalServicePrivate'=any(tags)) and id = {USER ID HERE};
```

**Review by ryanslade — APPROVED** [review]

**Pull request merged**